### PR TITLE
kenv 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by seivan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

Adding https://github.com/thisendout/kenv

### CODE REVIEW

**Comment by SMillerDev** [comment]
It depends on the audit you run. The documentation [here](https://docs.brew.sh/Manpage)  documents the `--new-formula` option (which should be used in this case.)

**Comment by seivan** [comment]
This is the one I am following: https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request 
Maybe it should have a link to the one you suggested, as it didn't occur me to search through that one as well.

File: Formula/kenv.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Kenv < Formula
+  desc "Environment file injector for Kubernetes resources"
+  homepage "https://github.com/thisendout/kenv"
+  url "https://github.com/thisendout/kenv/archive/v0.3.1.tar.gz"
+  sha256 "c245583aee13cb0e1b6ec44e7d16fe83455fbca2657b73f49f1d7328b8594204"
```
Comment (zbeekman) [inline Formula/kenv.rb]: Can you please insert a new line after the sha256?

File: Formula/kenv.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    bin_path = buildpath/"src/github.com/thisendout/kenv"
```
Comment (DomT4) [inline Formula/kenv.rb]: Use `dir` rather than `bin_path` here please.

File: Formula/kenv.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    bin_path.install Dir["*"]
+    cd bin_path do
+      system "go", "build", "-o", bin/"kenv", "."
```
Comment (DomT4) [inline Formula/kenv.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

File: Formula/kenv.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    end
+  end
+
+  test do
+    vars_file = "varstoinject.env"
```
Comment (DomT4) [inline Formula/kenv.rb]: `testpath/"varstoinject.env"`

File: Formula/kenv.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    deployment_file = "deployment.yml"
```
Comment (DomT4) [inline Formula/kenv.rb]: `testpath/"deployment.yml"`

File: Formula/kenv.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    (testpath/vars_file).write <<~END
```
Comment (DomT4) [inline Formula/kenv.rb]: And then just:
```ruby
vars_file.write <<~EOS
  key1=value1
  key2=value2
EOS
```

File: Formula/kenv.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      key1=value1
+      key2=value2
+    END
+
+    (testpath/deployment_file).write <<-EOS
```
Comment (DomT4) [inline Formula/kenv.rb]: And:
```ruby
deployment_file.write <<~EOS
```

File: Formula/kenv.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      name: nginx
+    spec:
+      template:
+        spec:
+          containers:
+          - name: nginx
+            image: nginx:latest
+    EOS
+
+    require "open3"
```
Comment (DomT4) [inline Formula/kenv.rb]: Please place the `require`s at the top of the block, thanks.

**Review by DomT4 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**